build-certificates-dir.sh

```diff
@@ -16,8 +16,6 @@ make install
 if [[ $PKG_NAME = osg-ca-certs ]]; then
   # OSG Specific stuff
   cd ../letsencrypt-certificates
-  sed -i 's/\.txt//g' Makefile
-  for x in *.txt; do mv "$x" "${x%.txt}"; done
```

Comment:
I get that you are removing this hack because you removed the .txt extensions in your PR against upstream.  And i get that for our purposes the .txt extension is not used for anything.  But are we the only consumers of letsencrypt?  If not, it seems we would potentially be breaking other downstream consumers, and in that case i wonder why we would expect them to take this change.

Reply:
Yup, certainly a fair point, and we'll see what Jim says. If this doesn't work for upstream, then we'll fork it or fold in all the LE stuff directly into this repo. It looks like that was your original intention with the `letsencrypt-certificates` directory anyway.

Reply:
Ok.  IMO it seems to me that a user-breaking change like this deserves its own PR.

Slipping this into a PR ("New Let's Encrypt intermediate CAs") that mentions nothing about this user-breaking change would make me a bit grumpy, either as an upstream maintainer or as a downstream user.

Reply:
I'm not sure I understand what you mean by "user-breaking change" here since this has no discernible effect on the users installing the packaging. Time's of the essence, though, so I'm not going to go through the effort of splitting up the PRs.

Reply:
> It looks like that was your original intention with the letsencrypt-certificates directory anyway.

Yeah, we can add items to the `letsencrypt-certificates` dir that are not present in our upstream cache letsencrypt-certificates tarball.  The main idea of the pem files in that dir though was to avoid an external download dependency at build-time (as described in the [update-pems.sh](https://github.com/opensciencegrid/osg-certificates/blob/master/letsencrypt-certificates/update-pems.sh) script).

I see that in cilogon/letsencrypt-certificates#3 you added the new lets-encrypt-r3.pem and lets-encrypt-r4.pem directly to the repo, which does eliminate the need to put them in the letsencrypt-certificates dir here in the osg-certificates repo; but that is different from the pattern otherwise seen in cilogon/letsencrypt-certificates.  It seems they intended it always to be a build-time download.  (Which is to say, i suspect we will want to download these and put them into the letsencrypt-certificates dir of our osg-certificates repo.)

Reply:
> I'm not sure I understand what you mean by "user-breaking change" here since this has no discernible effect on the users installing the packaging

Just to explain - what i meant by "user" was anyone who uses the cilogon/letsencrypt-certificates repo directly.  I used the word "consumer" earlier in the same sense:

>  But are we the only consumers of letsencrypt? If not, it seems we would potentially be breaking other downstream consumers

So i just meant that if anyone else had built packaging machinery around cilogon/letsencrypt-certificates that expected to see the .txt extensions (as we did), then cilogon/letsencrypt-certificates#3 might break that (as it would have for us, if we hadn't at the same time fixed our packaging machinery).

But it seems like cilogon was fine with it, which i guess is all that matters :)